engine.go

```diff
@@ -15,6 +16,8 @@ type (
 	Engine interface {
 		// Name of the engine. Ex. "wazero"
 		Name() string
+		NewWithMetering(code []byte, hostCallHandler HostCallHandler, maxInstructions uint64, pfn unsafe.Pointer) (Module, error)
```

Comment:
How about putting this in ModuleConfig instead as it can prevent api clash. Also, this sounds like configuration :D

Reply:
Agreed. Working on it.

Reply:
I am now using a completely different approach, but it's gone